service/src/main/java/org/jboss/sbomer/service/feature/sbom/service/SbomGenerationRequestRepository.java

```diff
@@ -0,0 +1,52 @@
+/**
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2023 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.sbomer.service.feature.sbom.service;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.transaction.Transactional;
+
+import org.jboss.sbomer.core.errors.NotFoundException;
+import org.jboss.sbomer.service.feature.sbom.model.Sbom;
+import org.jboss.sbomer.service.feature.sbom.model.SbomGenerationRequest;
+
+import io.quarkus.panache.common.Parameters;
+import lombok.extern.slf4j.Slf4j;
+
+@ApplicationScoped
+@Slf4j
+public class SbomGenerationRequestRepository extends RSQLBaseRepository<SbomGenerationRequest, String> {
+
+    protected Class getEntityClass() {
+        return SbomGenerationRequest.class;
+    }
+
+    @Transactional
+    public void deleteRequest(String id) {
+
+        log.info("Deleting SbomGenerationRequest with id: '{}' and all its associated Sboms", id);
+        SbomGenerationRequest request = findById(id);
+        if (request == null) {
+            throw new NotFoundException("Could not find any SBOM generation request with id '{}'", id);
+        }
+
+        Sbom.delete("generationRequest.id = :id", Parameters.with("id", id));
```

Comment:
Maybe we could add more logging here, to at least print how many entities were deleted?